src/services/submissionService.ts

```diff
@@ -101,7 +155,31 @@ export const mapAndValidateSpokenLanguages = (spokenLanguages: SpokenLanguageInp
     return validatedLanguages.map(gqlSpokenLanguageToDbSpokenLanguage)
 }
 
-const mapDbEntityTogqlEntity = (dbEntity: DocumentData) : Submission => {
+export const mapGqlFiltersToDbFilters = (filters: gqlType.SubmissionFilters):
+    dbSchema.SubmissionFilters => {
+    const mappedLanguages = filters.spokenLanguages?.map(lang => lang ? {
+        iso639_3: lang.iso639_3 || '',
+        nameJa: lang.nameJa || '',
+        nameEn: lang.nameEn || '',
+        nameNative: lang.nameNative || ''
+    } : undefined)
+
+    const filteredLanguages = (mappedLanguages || [])
+        .filter(lang => lang !== undefined) as dbSchema.SpokenLanguage[]
+
+    return {
+        ...filters,
+        googleMapsUrl: filters.googleMapsUrl === null ? undefined : filters.googleMapsUrl,
```

Comment:
You can use the `??` operator here too to simplify these mappings

```suggestion
        googleMapsUrl: !filters.googleMapsUrl ?? undefined,
```

Reply:
another note, `false` might be a better default for these booleans than undefined. Thoughts?